File: backend/services/core/src/main/java/app/mnema/core/media/config/MediaClientProps.java
package app.mnema.core.media.config;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "app.media")
public record MediaClientProps(
        String baseUrl,
        @NotBlank String internalToken
) {
}

File: backend/services/user/src/main/kotlin/app/mnema/user/media/config/MediaClientProps.kt
package app.mnema.user.media.config

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "app.media")
data class MediaClientProps(
    val baseUrl: String,
    @field:NotBlank val internalToken: String? = null
)
